### PR TITLE
Create a "TS basics" tut

### LINKED ISSUE #25: Adapt content from BDI tskit workshop
Here: https://github.com/mcveanlab/tskit-workshop

@hyanwong do you think it's worth pulling together bits of these notebooks into an "intro to tskit" page?

### PR DESCRIPTION
**Pull request by hyanwong**
This provides some definitions, etc and I think is a useful initial tutorial page, to be read after "what is a tree sequence" but before actually getting stuck in to python code and algorithms. It's largely taken from the BDI workshop notebook, hence closes #25

### CODE REVIEW

File: basics.md [hunk 1]
```diff
@@ -0,0 +1,345 @@
...
+### Edges
+
+In an evolutionary tree, the connections between nodes are conventionally referred to as
+"branches". In a tree sequence, we instead refer to directional connections between nodes
+as _edges_, emphasizing that they usually persist across multiple trees. Each edge is
+associated with a parent node ID and a child node ID. The time of the parent node must be
+strictly greater than the time of the child node, and the difference in these times is
+sometimes referred to as the "length" of the edge. Since trees in a tree sequence are
+usually taken to represent marginal trees along a genome, as well as the time dimension
+heach edge also has a genomic _span_, defined by the edge's *left* and a *right* position
```
Comment (benjeffery) [inline basics.md:133]:
```suggestion
each edge also has a genomic _span_, defined by the edge's *left* and a *right* position
```

File: basics.md [hunk 2]
```diff
@@ -0,0 +1,345 @@
...
+sequences that have been inferred from real data, it is possible to have a parent node
+with 3 or more children in a particular tree. Such multifucations are known as
+*polytomies*.
+:::
+
+### Individuals and populations
+
+Sometimes we know that one or more nodes belong to a particular _individual_. For example,
+most organisms are diploid, so if we sequence the DNA from an individual we are likely to
+obtain two copies of each autosomal chromosome. The tree above could therefore represent
```
Comment (benjeffery) [inline basics.md:183]:
```suggestion
obtain two copies of each autosomal chromosome. The tree with six sample nodes above could therefore represent
```

File: basics.md [hunk 3]
```diff
@@ -0,0 +1,345 @@
...
+this:
+
+```{code-cell} ipython3
+print(f"{ts.num_mutations} mutation(s) in the tree sequence:")
+for mutation in ts.mutations():
+    print("*", mutation)
+```
+
+The mutation can have a {attr}`~Mutation.time` or if, as in this case, the times of
+mutations in the tree sequence are unknown, all mutations can have the special value
```
Comment (benjeffery) [inline basics.md:240]:
```suggestion
mutations in the tree sequence are unknown, all mutations can have the special NaN value
```

File: basics.md [hunk 4]
```diff
@@ -0,0 +1,345 @@
...
+while the others have the {attr}`~Site.ancestral_state` of "T". This means that Ada is
+homozygous for "T", Bob is heterozygous "T|G", and Cat is homozygous for "G".
+In other words the ancestral state and the details of any mutations at that site,
+when coupled with the tree topology at the site {attr}`~Site.position` is sufficient to
+define the allelic state possessed by each sample.
+
+Note that even though the genome is 1000 base pairs long, the tree sequence only contains
+a single site, because we usually only bother defining *variable* sites in a tree
+sequence (e.g. positions seen in studies to have samples possessing different alleles at
+that genomic location). It is perfectly possible to have a site with no mutations
```
Comment (benjeffery) [inline basics.md:261]:
```suggestion
that genomic location). It is perfectly possible to have a site with no mutations or silent mutations
```

File: basics.md [hunk 5]
```diff
@@ -0,0 +1,345 @@
...
+ultimate coalesence happened in a different ancestor (this is easy to see in a simulation,
+but hard to detect in real data). The SPR that created the third tree must involve pruning
+either the subtree below node 7 or the subtree below node 9 and grafted it higher up on
+the same lineage. In this case it turns out to be impossible to tell just from the tree
+sequence which of those two SPRs actually happened in the simulation.
+
+In general, each detectable recombination occurring in ancestral history results in a
+single SPR in the tree sequence. If recombination breakpoints occurs at unique
+positions (an "infinite sites" model of breakpoints), then the number of trees in a tree
+sequence equals the number of ancestral recombination events minus one. If recombinations
```
Comment (benjeffery) [inline basics.md:300]: I'm either missing something here or this should be that the number of trees is equal to the number of recombinations plus one? :confused:

File: basics.md [hunk 6]
```diff
@@ -0,0 +1,345 @@
...
+
+:::{todo}
+Link to the ARG tutorial: https://github.com/tskit-dev/tutorials/issues/43
+:::
+
+
+### Tables
+
+:::{todo}
+The underlying data in a tree sequence is best thought of as a set of tables. The API
```
Comment (benjeffery) [inline basics.md:340]:
```suggestion
The underlying data in a tree sequence is stored as a set of tables. The API
```

**Review by benjeffery — APPROVED** [review]
This is really good @hyanwong, just a couple of quibbles.

File: basics.md (hunk 5 above)
Comment (hyanwong) [reply basics.md:300]: Well spotted. Copied from the old workshop material, where it's presumably wrong too!

**Comment by hyanwong** [comment]
> This is really good @hyanwong, just a couple of quibbles.

Cheers @benjeffery - all suggestions added and a few more tidies.

**Review by jeromekelleher — APPROVED** [review]

**Comment by hyanwong** [comment]
Thanks both. If we could merge this I can ping Konrad who is looking for material (I don't thing automerge is set up on the tutorials repo yet.

**Pull request merged**